### PR TITLE
Refactor metadata search

### PR DESCRIPTION
**Pull request by oschwald**
The performance on this does not differ from the previous
implementation, but it is easier to follow.

If we wanted to improve performance, we would probably read in bigger
chunks at a time and then search within each chunk.

### CODE REVIEW

File: src/MaxMind/Db/Reader.php [hunk 1]
```diff
@@ -265,19 +265,18 @@ private function findMetadataStart($filename)
...
-
-        for ($i = 0; $i <= $metadataMaxLengthExcludingMarker; ++$i) {
-            for ($j = 0; $j < $markerLength; ++$j) {
-                fseek($handle, $fileSize - $i - $j - 1);
-                $matchBit = fgetc($handle);
-                if ($matchBit !== $marker[$markerLength - $j - 1]) {
-                    continue 2;
-                }
+
+        $minStart = $fileSize - min(self::$METADATA_MAX_SIZE + 1, $fileSize);
```
Comment (horgh) [inline src/MaxMind/Db/Reader.php]: Why +1?

File: src/MaxMind/Db/Reader.php [hunk 2]
```diff
@@ -265,19 +265,18 @@ private function findMetadataStart($filename)
...
+
+        for ($offset = $fileSize - $markerLength - 1; $offset >= $minStart; --$offset) {
```
Comment (horgh) [inline src/MaxMind/Db/Reader.php]: Why -1? I guess in this case the metadata will never be just the marker though so it probably doesn't matter.

File: src/MaxMind/Db/Reader.php (hunk 1 above)
Comment (oschwald) [reply src/MaxMind/Db/Reader.php]: Because the offsets are 0 indexed so the last offset would be `$fileSize - 1`.

Comment (oschwald) [reply src/MaxMind/Db/Reader.php]: Sorry, you are right. I'll change it tomorrow.

**Pull request merged**